server/lib/dark_worlds_server/engine/bot_player.ex

```diff
@@ -200,27 +191,100 @@ defmodule DarkWorldsServer.Engine.BotPlayer do
     {x, -y}
   end
 
-  def decide_objective(%{bots_enabled: false}, _bot_id) do
-    :nothing
+  def decide_objective(bot_state, %{bots_enabled: false}, _bot_id, _closest_entity) do
+    Map.put(bot_state, :objective, :nothing)
   end
 
-  def decide_objective(%{myrra_state: myrra_state}, bot_id) do
+  def decide_objective(bot_state, %{game_state: %{myrra_state: myrra_state}}, bot_id, closest_entity) do
     bot = Enum.find(myrra_state.players, fn player -> player.id == bot_id end)
 
+    objective =
+      case bot do
+        nil ->
+          :waiting_game_update
+
+        bot ->
+          out_of_area? = Enum.any?(bot.effects, fn {k, _v} -> k == :out_of_area end)
+
+          cond do
+            out_of_area? ->
+              :flee_from_zone
+
+            not Enum.empty?(closest_entity) ->
+              :attack_enemy
+
+            true ->
+              :random_movement
+          end
+      end
+
+    Map.put(bot_state, :objective, objective)
+  end
+
+  def decide_objective(bot_state, _, _, _), do: Map.put(bot_state, :objective, :nothing)
+
+  defp get_closes_entity(%{myrra_state: game_state}, bot_id) do
+    # TODO maybe we could add a priority to the entities.
+    # e.g. if the bot has low health priorities the loot boxes
+    bot = Enum.find(game_state.players, fn player -> player.id == bot_id end)
+
     case bot do
       nil ->
-        :waiting_game_update
+        %{}
 
       bot ->
-        out_of_area? = Enum.any?(bot.effects, fn {k, _v} -> k == :out_of_area end)
+        players_distances =
+          game_state.players
+          |> Enum.filter(fn player -> player.status == :alive and player.id != bot.id end)
+          |> map_entities(bot, :enemy)
+
+        loots_distances =
+          game_state.loots
+          |> map_entities(bot, :loot)
 
-        if out_of_area? do
-          :flee_from_zone
-        else
-          Enum.random([:attack_enemy, :random_movement])
+        cond do
+          Enum.empty?(loots_distances) and Enum.empty?(players_distances) ->
+            %{}
+
+          Enum.empty?(loots_distances) ->
+            hd(players_distances)
+
+          Enum.empty?(players_distances) ->
+            hd(loots_distances)
+
+          true ->
+            Enum.min_by([hd(loots_distances), hd(players_distances)], fn entity -> entity.distance_to_entity end)
         end
     end
   end
 
-  def decide_objective(_, _), do: :nothing
+  defp get_closes_entity(_, _) do
+    %{}
+  end
```

Comment:
😛 
(Beware the typo is everywhere)
```suggestion
  defp get_closest_entity(_, _) do
    %{}
  end
```

Reply:
whoops, should be fixed now